src/parquet/column/statistics-test.cc

```diff
@@ -0,0 +1,158 @@
+    statistics1.Update(values_ptr_, values_.size(), 0);
+    std::string encoded_min = statistics1.EncodedMin();
+    std::string encoded_max = statistics1.EncodedMax();
+
+    TypedRowGroupStatistics<TestType> statistics2(
+        schema_.get(), encoded_min, encoded_max, values_.size(), 0, 0);
+
+    ASSERT_EQ(encoded_min, statistics2.EncodedMin());
+    ASSERT_EQ(encoded_max, statistics2.EncodedMax());
+    ASSERT_EQ(statistics1.Min(), statistics2.Min());
+    ASSERT_EQ(statistics1.Max(), statistics2.Max());
+  }
+
+  void TestReset() {
+    GenerateData(1000);
+
+    TypedRowGroupStatistics<TestType> statistics(schema_.get());
+    statistics.Update(values_ptr_, values_.size(), 0);
+    ASSERT_EQ(values_.size(), statistics.NumValues());
+
+    statistics.Reset();
+    ASSERT_EQ(0, statistics.NullCount());
+    ASSERT_EQ(0, statistics.NumValues());
+    ASSERT_EQ("", statistics.EncodedMin());
+    ASSERT_EQ("", statistics.EncodedMax());
+  }
+
+  void TestMerge() {
+    int num_null[2];
+    random_numbers(2, 42, 0, 100, num_null);
+
+    TypedRowGroupStatistics<TestType> statistics1(schema_.get());
+    GenerateData(1000);
+    statistics1.Update(values_ptr_, values_.size() - num_null[0], num_null[0]);
+
+    TypedRowGroupStatistics<TestType> statistics2(schema_.get());
+    GenerateData(1000);
+    statistics2.Update(values_ptr_, values_.size() - num_null[1], num_null[1]);
+
+    TypedRowGroupStatistics<TestType> total(schema_.get());
+    total.Merge(statistics1);
+    total.Merge(statistics2);
+
+    ASSERT_EQ(num_null[0] + num_null[1], total.NullCount());
+    ASSERT_EQ(values_.size() * 2 - num_null[0] - num_null[1], total.NumValues());
+    ASSERT_EQ(total.Min(), std::min(statistics1.Min(), statistics2.Min()));
+    ASSERT_EQ(total.Max(), std::max(statistics1.Max(), statistics2.Max()));
+  }
+
+ protected:
+  std::vector<T> values_;
+  std::vector<uint8_t> buffer_;
+  std::vector<uint8_t> bool_buffer_;
+  T* values_ptr_;
+
+ private:
+  NodePtr node_;
+  std::shared_ptr<ColumnDescriptor> schema_;
+};
+
+// FIXME code duplication with column-writer-test.cc
+template <typename TestType>
+void TestRowGroupStatistics<TestType>::GenerateData(int64_t num_values) {
+  values_.resize(num_values);
+  InitValues<T>(num_values, values_, buffer_);
+  values_ptr_ = values_.data();
+}
+
+template <>
+void TestRowGroupStatistics<BooleanType>::GenerateData(int64_t num_values) {
+  values_.resize(num_values);
+  InitValues<T>(num_values, values_, buffer_);
+  bool_buffer_.resize(num_values);
+  std::copy(values_.begin(), values_.end(), bool_buffer_.begin());
+  values_ptr_ = reinterpret_cast<bool*>(bool_buffer_.data());
+}
+
+using TestTypes = ::testing::Types<Int32Type, Int64Type, Int96Type, FloatType, DoubleType,
+    ByteArrayType, FLBAType, BooleanType>;
+
+TYPED_TEST_CASE(TestRowGroupStatistics, TestTypes);
+
+TYPED_TEST(TestRowGroupStatistics, MinMaxEncode) {
+  this->TestMinMaxEncode();
```

Comment:
that might be too ambitious for this patch, can we create a JIRA for it to track? C++11 now does have some built in library support for unicode (see the `<locale>` include) but it could end up being necessary to pull in ICU or some 3rd party C++ unicode library